components/tidb_query_expr/src/impl_cast.rs

```diff
@@ -919,6 +919,9 @@ fn cast_int_as_duration(
                 if err.is_overflow() {
                     ctx.handle_overflow_err(err)?;
                     Ok(None)
+                } else if err.is_truncated() {
```

Comment:
Could you please add a `test_int_as_duration` for this function?

You can take `test_int_as_time` as an example.

Reply:
rest LGTM